src/igraph/__init__.py

```diff
@@ -3190,6 +3190,54 @@ def DataFrame(klass, edges, directed=True, vertices=None, use_vids=False):
 
         return g
 
+    def get_vertex_dataframe(self):
+        """Export vertices with attributes to pandas.DataFrame
+
+        @return: a pandas.DataFrame representing vertices and their attributes.
+          If the vertices have names, the index of the dataframe contains them,
+          otherwise it's integers from 0 up (which are also the vertex IDs).
+          The order of the vertices is the same as in the graph, so the n-th
+          row describes the vertex with ID i.
+        """
+        import pandas as pd
+
+        df = pd.DataFrame(
+            {attr: self.vs[attr] for attr in self.vertex_attributes()},
+            index=list(range(self.vcount())),
+        )
+        if 'name' in df.columns:
+            df.set_index('name', inplace=True)
+        return df
+
+    def get_edge_dataframe(self):
```

Comment:
Here, again, we might make it optional for the user to use either integer vertex indices, or the vertex labels (in the `source` and `target` columns).

Reply:
as above, I'd be more in favour of educating the user via docstrings than setting up a bunch of if/else for one-liners. Especially if that involves multiindices.

Reply:
I agree with @iosonofabio here if that matters

Reply:
Frankly, getting the source and target names are five lines of code, which I don't consider entirely trivial for end users. I personally never had too much trouble with `MultiIndex`, but maybe I just got lucky. But good if this is in the documentation.